pkg/trace/obfuscate/sql_tokenizer.go

```diff
@@ -480,60 +493,68 @@ func (tkn *SQLTokenizer) scanString(delim rune, kind TokenKind) (TokenKind, []by
 }
 
 func (tkn *SQLTokenizer) scanCommentType1(prefix string) (TokenKind, []byte) {
-	buffer := &bytes.Buffer{}
-	buffer.WriteString(prefix)
 	for tkn.lastChar != EOFChar {
 		if tkn.lastChar == '\n' {
-			tkn.consumeNext(buffer)
+			tkn.next()
 			break
 		}
-		tkn.consumeNext(buffer)
+		tkn.next()
 	}
-	return Comment, buffer.Bytes()
+	return Comment, tkn.grab()
 }
 
 func (tkn *SQLTokenizer) scanCommentType2() (TokenKind, []byte) {
-	buffer := &bytes.Buffer{}
-	buffer.WriteString("/*")
 	for {
 		if tkn.lastChar == '*' {
-			tkn.consumeNext(buffer)
+			tkn.next()
 			if tkn.lastChar == '/' {
-				tkn.consumeNext(buffer)
+				tkn.next()
 				break
 			}
 			continue
 		}
 		if tkn.lastChar == EOFChar {
 			tkn.setErr("unexpected EOF in comment")
-			return LexError, buffer.Bytes()
+			return LexError, tkn.grab()
 		}
-		tkn.consumeNext(buffer)
+		tkn.next()
 	}
-	return Comment, buffer.Bytes()
+	return Comment, tkn.grab()
 }
 
-func (tkn *SQLTokenizer) consumeNext(buffer *bytes.Buffer) {
-	if tkn.lastChar == EOFChar {
-		// This should never happen.
-		panic("unexpected EOF")
+// Next sets tkn.lastChar to the next rune at tkn.buf[tkn.index], and advances the tkn.pos and
+// tkn.index by its size. If next() is unable to scan a character, tkn.lastChar is set to EOFChar.
+func (tkn *SQLTokenizer) next() {
+	ch, n := utf8.DecodeRune(tkn.buf[tkn.index:])
+	if ch == utf8.RuneError && n == 0 {
+		// only EOF is possible
+		tkn.pos++
+		tkn.lastChar = EOFChar
+		return
 	}
-	buffer.WriteRune(tkn.lastChar)
-	tkn.next()
-}
 
-func (tkn *SQLTokenizer) next() {
-	ch, _, err := tkn.rd.ReadRune()
+	len := utf8.RuneLen(ch)
 	if tkn.lastChar != 0 || tkn.pos > 0 {
 		// we are past the first character
-		tkn.pos += utf8.RuneLen(ch)
-	}
-	if err != nil {
-		// only EOF is possible
-		tkn.lastChar = EOFChar
-	} else {
-		tkn.lastChar = ch
+		tkn.pos += len
 	}
+	tkn.index += len
+	tkn.lastChar = ch
+}
+
+// grab returns a byte slice of all the characters advanced over with next() since the last call to
+// grab.
+func (tkn *SQLTokenizer) grab() []byte {
```

Comment:
Could `bytes` be a good name for this method?
```go
// bytes returns all the bytes that were advanced over since its last call.
```

Reply:
`bytes` will be better I think.